binr/rahash2/rahash2.c

```diff
@@ -288,10 +288,80 @@ int is_power_of_two(const ut64 x) {
 	return x && !(x & (x - 1));
 }
 
+int encrypt_or_decrypt (const char *algo, bool to_encrypt, const char *hashstr, int hashstr_len) {
+	bool no_key_mode = !strcmp ("base64", algo) || !strcmp ("base91", algo); //TODO: generalise this for all non key encoding/decoding.
```

Comment:
@radare if I do a check regarding `crypto->h->set_key != NULL`, then that clear the `no_key_mode` from line 296 `if (no_key_mode || r_crypto_set_key (cry, s.buf, s.len, 0, 0)) {` but then how should I tackle with the line 293 `if (no_key_mode || s.len > 0) {`? If this can be done, then just adding the simple check that you mentioned would be great. I will look by this too and add comment if I get something :)